kong/pdk/client.lua

```diff
@@ -199,6 +200,63 @@ local function new(self)
   end
 
 
+  ---
+  -- Returns the Nginx subsystem used by the client. It can be
+  -- "http" or "stream"
+  --
+  -- @function kong.client.get_subsystem
+  -- @phases access, header_filter, body_filter, log
+  -- @treturn string a string with either `"http"` or `"stream"`
+  -- @usage
+  -- kong.client.get_subsystem() -- "http"
+  function _CLIENT.get_subsystem()
+    check_phase(AUTH_AND_LATER)
```

Comment:
We should be able to call this API from `init_by_lua`, `init_worker_by_lua` as well, which further highlights the need for this API to be a root API.